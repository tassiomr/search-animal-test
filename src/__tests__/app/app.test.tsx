import App from '@app/app';
import { render } from '@testing-library/react';

describe('App Test Suite', () => {
  it('should render correct app', () => {
    const screen = render(<App />);

    expect(screen).toMatchSnapshot();
  });
});
