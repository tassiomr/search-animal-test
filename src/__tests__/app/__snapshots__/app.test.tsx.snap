// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Suite should render correct app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-container"
        data-testid="home-page"
      >
        <div
          class="header-container"
          data-testid="home-page-header"
        >
          <p
            class="text-title-home-page"
          >
            <span>
              AgileContent
               
            </span>
             
            FrontEnd Test
          </p>
          <div
            class="avatar-icon-container"
          >
            <svg
              class="icon-google-apps"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
              />
            </svg>
            <img
              class="avatar"
              data-testid="avatar"
              src="https://i.scdn.co/image/ab6761610000e5eb4a61c7615c8125e832ffc32c"
            />
          </div>
        </div>
        <div
          class="home-body"
          data-testid="home-page-body"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <div
            class="spacer"
          />
          <div
            class="search-component-container"
          >
            <div
              class="input-container"
            >
              <svg
                class="icons-search-component"
                data-testid="search-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <input
                data-testid="input-search-component"
                value=""
              />
              <svg
                class="icons-search-component--none"
                data-testid="clean-icon-button"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </div>
          </div>
          <div
            class="spacer"
          />
          <button
            data-testid="home-page-button"
            disabled=""
          >
            Buscar
          </button>
        </div>
        <div
          class="footer-container"
          data-testid="home-page-footer"
        >
          <p
            class="text-footer"
          >
            ⓒ Google 2023
          </p>
          <p
            class="text-footer"
          >
            vesion: 2
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-container"
      data-testid="home-page"
    >
      <div
        class="header-container"
        data-testid="home-page-header"
      >
        <p
          class="text-title-home-page"
        >
          <span>
            AgileContent
             
          </span>
           
          FrontEnd Test
        </p>
        <div
          class="avatar-icon-container"
        >
          <svg
            class="icon-google-apps"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
            />
          </svg>
          <img
            class="avatar"
            data-testid="avatar"
            src="https://i.scdn.co/image/ab6761610000e5eb4a61c7615c8125e832ffc32c"
          />
        </div>
      </div>
      <div
        class="home-body"
        data-testid="home-page-body"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <div
          class="spacer"
        />
        <div
          class="search-component-container"
        >
          <div
            class="input-container"
          >
            <svg
              class="icons-search-component"
              data-testid="search-icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              data-testid="input-search-component"
              value=""
            />
            <svg
              class="icons-search-component--none"
              data-testid="clean-icon-button"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
        <div
          class="spacer"
        />
        <button
          data-testid="home-page-button"
          disabled=""
        >
          Buscar
        </button>
      </div>
      <div
        class="footer-container"
        data-testid="home-page-footer"
      >
        <p
          class="text-footer"
        >
          ⓒ Google 2023
        </p>
        <p
          class="text-footer"
        >
          vesion: 2
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
